frontend/src/pages/modelCatalog/utils.ts

```diff
@@ -0,0 +1,38 @@
+import { CatalogModel, ModelCatalogSource } from '~/concepts/modelCatalog/types';
+import { ModelDetailsRouteParams } from './const';
+
+export const findModelFromModelCatalogSources = (
+  modelCatalogSources: ModelCatalogSource[],
+  source: string | undefined,
+  repositoryName: string | undefined,
+  modelName: string | undefined,
+  tag: string | undefined,
+): CatalogModel | null => {
+  const modelCatalogSource = modelCatalogSources.find((mcSource) => mcSource.source === source);
+  if (!modelCatalogSource) {
+    return null;
+  }
+
+  const modelMatched = modelCatalogSource.models.find(
+    (m) =>
+      m.repository === repositoryName &&
+      tag &&
+      m.name === modelName &&
+      m.artifacts?.some((f) => f.tags?.includes(tag)),
+  );
+
+  return modelMatched || null;
+};
+
+export const encodeParams = (params: ModelDetailsRouteParams): ModelDetailsRouteParams =>
+  Object.fromEntries(
+    Object.entries(params).map(([key, value]) => [
+      key,
+      encodeURIComponent(value).replace(/\./g, '%252E'),
```

Comment:
just curious why this needs the extra encoding of that replace

Reply:
Hmm, yeah @ppadti I'm also curious about this - we don't need to block merge, but maybe we can follow up with a loose PR if necessary

Reply:
yeah, I did this because `:tag ` in route param has decimal value and tag being the last route param, I was getting 404 error. Even when I encode `1.3-1732870892` to `1%2E3-1732870892`, after pasting this on url  - the browser automatically decoded that. So thought of doing double encoding. But after checking this with @manaswinidas I got to know that, it was just me who got that error. I am not sure why. Sorry, I can update it with a follow up PR.

<img width="1507" alt="Screenshot 2025-02-13 at 5 21 41 PM" src="https://github.com/user-attachments/assets/82ed87af-2a23-47e1-a072-11019df5e8ce" />